test/integration/consul-container/upgrade/basic_test.go

```diff
@@ -0,0 +1,324 @@
+package consul_container
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+
+	consulCluster "github.com/hashicorp/consul/integration/consul-container/libs/consul-cluster"
+	consulNode "github.com/hashicorp/consul/integration/consul-container/libs/consul-node"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+
+	"github.com/stretchr/testify/require"
+)
+
+var curImage = flag.String("uut-version", "local", "docker image to be used as UUT (unit under test)")
+var latestImage = flag.String("latest-version", "latest", "docker image to be used as latest")
+
+const retryTimeout = 10 * time.Second
+const retryFrequency = 500 * time.Millisecond
+
+func TestLatestGAServersWithCurrentClients(t *testing.T) {
```

Comment:
Could we add doc strings for the intention of each test? I find it helpful for future proofing and it helps me understand the goal of test better 🤓